Readme.md
# Identity 2

This addon allows you to specify your main character's name, an identity format string, and which channels will have your identity tag prepended. It also allows you to specify a nickname for use in Raid and Party channels.


## Basic instructions:

use `/id` to go the configuration window

Enable
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Active/Deactive Identity2

    
Fun
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Enable/Disable Fun mode in special days
    
Format
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The base for witch the Identity will be generated on the chat. The tokens you can use are:
 - %s -> Will be replaced by the appropriate identity
 - %z -> Will be replaced by the name of the current zone
 - %l -> Will be replaced by the character level
 - %g -> Will be replaced by the character guild
 - %r -> Will be replaced by the realm name.
 - %f -> Will be replaced by the character faction

Any other character will be kept unchanged, for example the format:

[%s-%r]<%g>(%l) - %z

will look like this:

[Wall-Nesingwary]<of The Queue>(100) - Nagrand

## Contact

Lavindar
philipi_will@hotmail.com

Wall
Guild: of The Queue 
Server: Nesingwary-US

## Credits

Currently maintained by Lavindar (philipi_will@hotmail.com)
Guild: of The Queue
Server: Nesingwary-US

----

Former author Kjallstrom (ultranurd@gmail.com)
Guild: Mellonea
Server: Kirin Tor

----

Original IDENTITY by Ferusnox
Guild: Heaven and Earth
Server: Cenarion Circle
"Just call me Nox"


changelog.txt
====Version 4.2.4

* Added faction token

====Version 4.2.3

* TOC bump

====Version 4.2.2

* TOC bump

====Version 4.2.1

* TOC bump

====Version 4.2.0

* TOC bump
* Added a check to avoid sending messages bigger than the maximun alowed after adding the identiy text

====Version 4.1.0

* TOC bump
* Added support for Battle.net Whispers

====Version 4.0.3

TOC bump

====Version 4.0.2

TOC bump

====Version 4.0.1

Aparently I rewrote the whole addon and still forgot to change the TOC version :)

====Version 4.0.0

* Configuration are now account wide with profile support
* MainName and NickName deprecated in favor of a single Identity field, with option in each channel to have a channel specific Identity
* Moved all configurations to a GUI instead of using commands. Any old command will open the new GUI
* Rewrite of the code in preparation for Battle for Azeroth Communities support

====Version 3.4.1

Fixed wrong value

====Version 3.4.0

TOC bump

====Version 3.3.4

Definitely not April's Fool stuff

====Version 3.3.3

* TOC bump

====Version 3.3.2

* TOC bump

====Version 3.3.1

* TOC bump

====Version 3.3.0

* Added support for the Instance channel

====Version 3.2.0

* code optimization for future GUI
* zone command removed, for zone info in the identity the token %z should be used

====Version 3.1.0

* TOC bump

====Version 3.0.1

* updated the readme file
* improved the way to format the Identity string, added tokens to set the zone, level, guild, and realm.
* added a check for new configurations on version change
* added message slash command to define the load message behavior
* improved the help command
* zone command is now obsolete, but remains functional for backwards compatibility

===Version 2.2-60100

* updated ToC to 6.1


Identity.lua
-----
-- Identity Addon
-- Heavily modified by Kjallstrom, Mellonea, Kirin Tor
-- Created by Ferusnox, Heaven and Earth, Cenarion Circle
-- Inspired by Thelma Incognito Addon
-----

-----
-- INITIALIZATION
-----

-- Sets the current Identity version
local Identity_VERSION = "2.0-30000";

-- Stores the unmodified chat message
local Identity_OriginalSendChatMessage;

-- Called when Identity is loaded at UI loadtime
function Identity_OnLoad()
    -- Prepare to read saved variables
    this:RegisterEvent("VARIABLES_LOADED");

    -- Register slash commands
    SlashCmdList["IDENTITY"] = Identity_Cmd;
    SLASH_IDENTITY1 = "/identity";
    SLASH_IDENTITY2 = "/id";
end

-- Handle the variable load event
function Identity_OnEvent()
    if (event == "VARIABLES_LOADED") then
        -- Check if this is the first time Identity has been loaded
        if (not IdentitySettings) then
            -- Set the defaults
            Identity_InitSettings();
        end

        -- Intercept chat events
        Identity_OriginalSendChatMessage = SendChatMessage;
        SendChatMessage = Identity_SendChatMessage;

        -- Indicate that Identity is done loading
        DEFAULT_CHAT_FRAME:AddMessage("Identity " .. Identity_VERSION .. " loaded", 0.4, 0.4, 1.0);
    end
end

-- Create a fresh, default Identity configuration
function Identity_InitSettings()
    IdentitySettings = {};
    IdentitySettings.Enabled = true;
    IdentitySettings.Format = "[%s]";
    IdentitySettings.MainName = "";
    IdentitySettings.NickName = "";
    IdentitySettings.DisplayZone = false;
    IdentitySettings.Channels = {};
    IdentitySettings.Channels.Guild = false;
    IdentitySettings.Channels.Officer = false;
    IdentitySettings.Channels.Raid = false;
    IdentitySettings.Channels.Party = false;
    IdentitySettings.Channels.Tell = false;
    IdentitySettings.Channels.C01 = false;
    IdentitySettings.Channels.C02 = false;
    IdentitySettings.Channels.C03 = false;
    IdentitySettings.Channels.C04 = false;
    IdentitySettings.Channels.C05 = false;
    IdentitySettings.Channels.C06 = false;
    IdentitySettings.Channels.C07 = false;
    IdentitySettings.Channels.C08 = false;
    IdentitySettings.Channels.C09 = false;
    IdentitySettings.Channels.C10 = false;
end

-----
-- CHAT MESSAGE HANDLING
-----

function Identity_SendChatMessage(msg, system, language, channel)
    -- Check if Identity is enabled
    if (IdentitySettings.Enabled) then
        -- Check if the main Identity is configured
        if (IdentitySettings.MainName ~= "") then
            -- Get the current Identity
            local main = Identity_GenerateMainName();

            -- Modify the message
            local newmsg = main .. " " .. msg;

            -- Guild channel
            if (IdentitySettings.Channels.Guild and system == "GUILD") then
                Identity_OriginalSendChatMessage(newmsg, system, language, channel);
                return;
            end

            -- Officer channel
            if (IdentitySettings.Channels.Officer and system == "OFFICER") then
                Identity_OriginalSendChatMessage(newmsg, system, language, channel);
                return;
            end

            -- Whispers
            if (IdentitySettings.Channels.Tell and system == "WHISPER") then
                Identity_OriginalSendChatMessage(newmsg, system, language, channel);
                return;
            end

            -- Numbered channels
            if (system == "CHANNEL") then
                local chanID = string.format("C%02d", channel);
                if (IdentitySettings.Channels[chanID]) then
                    Identity_OriginalSendChatMessage(newmsg, system, language, channel);
                    return;
                end
            end
       end

        -- Check if the nickname Identity is configured
        if (IdentitySettings.NickName ~= "") then
            -- Get the current Identity
            local nick = Identity_GenerateNickName();

            -- Modify the message
            local newmsg = nick .. " " .. msg;

            -- Raid channel
            if (IdentitySettings.Channels.Raid and (system == "RAID" or system == "BATTLEGROUND")) then
                Identity_OriginalSendChatMessage(newmsg, system, language, channel);
                return;
            end

            -- Party channel
            if (IdentitySettings.Channels.Party and system == "PARTY") then
                Identity_OriginalSendChatMessage(newmsg, system, language, channel);
                return;
            end
       end
    end

    -- Pass the message through unchanged
    Identity_OriginalSendChatMessage(msg, system, language, channel);
end

-----
-- COMMAND HANDLING
-----

-- Handle Identity commands
function Identity_Cmd(msg)
    -- Extract the command and arguments from the message
    local cmd, options = Identity_ParseCmd(msg);

    -- Check which command was specified, if any
    if (cmd == "" or cmd == "config") then
        Identity_PrintConfig();
    elseif (cmd == "help") then
        Identity_PrintHelp(options);
    elseif (cmd == "on") then
        Identity_Enable(true);
    elseif (cmd == "off") then
        Identity_Enable(false);
    elseif (cmd == "main") then
        IdentitySettings.MainName = Identity_CheckName(IdentitySettings.MainName, options);
    elseif (cmd == "nick") then
        IdentitySettings.NickName = Identity_CheckName(IdentitySettings.NickName, options);
    elseif (cmd == "enable") then
        Identity_EnableChannels(options, true);
    elseif (cmd == "disable") then
        Identity_EnableChannels(options, false);
    elseif (cmd == "zone") then
        Identity_SetZoneDisplay(options);
    elseif (cmd == "format") then
        Identity_SetFormat(options);
    elseif (cmd == "reset") then
        DEFAULT_CHAT_FRAME:AddMessage("Identity configuration cleared", 0.4, 0.4, 1.0);
        Identity_InitSettings();
        Identity_PrintConfig();
    else
        DEFAULT_CHAT_FRAME:AddMessage("Invalid Identity command '" .. msg .. "'", 1.0, 0.4, 0.4);
    end
end

-- Split a command into two parts: the name of the command, and any
-- command arguments.
function Identity_ParseCmd(msg)
    -- Ignore null messages
    if (msg) then
        -- Split the command and its arguments
        local s, e, cmd = string.find(msg, "(%S+)");
        if (s) then
            -- Command plus any options
            return cmd, string.sub(msg, e + 2);
        else
            -- All whitespace?
            return "";
        end
    end
end

-- Dumps the current Identity configuration
function Identity_PrintConfig()
    -- Print general Identity information
    DEFAULT_CHAT_FRAME:AddMessage("----", 1.0, 1.0, 1.0);
    DEFAULT_CHAT_FRAME:AddMessage("Identity " .. Identity_VERSION .. " configuration:", 0.4, 0.4, 1.0);
    if (IdentitySettings.Enabled) then
        DEFAULT_CHAT_FRAME:AddMessage("  Enabled", 0.4, 1.0, 0.4);
    else
        DEFAULT_CHAT_FRAME:AddMessage("  Disabled", 1.0, 0.4, 0.4);        
    end

    -- Check if the main Identity is configured
    if (IdentitySettings.MainName ~= "") then
        -- Get an example of the current Identity
        local main = Identity_GenerateMainName();

        -- Get the list of channels enabled for the current Identity
        local mainChannels = Identity_GetMainChannels();

        -- Display the main Identity
        DEFAULT_CHAT_FRAME:AddMessage("  Main name: " .. main, 0.4, 0.4, 1.0);
        if (mainChannels ~= "") then
            DEFAULT_CHAT_FRAME:AddMessage("    " .. mainChannels, 0.4, 1.0, 0.4);
        else
            DEFAULT_CHAT_FRAME:AddMessage("    All channels disabled", 1.0, 0.4, 0.4);
        end
    else
        DEFAULT_CHAT_FRAME:AddMessage("  No main name", 1.0, 0.4, 0.4);
    end

    -- Check if the nickname Identity is configured
    if (IdentitySettings.NickName ~= "") then
        -- Get an example of the current Identity
        local nick = Identity_GenerateNickName();

        -- Get the list of channels enabled for the current Identity
        local nickChannels = Identity_GetNickChannels();

        -- Display the nick Identity
        DEFAULT_CHAT_FRAME:AddMessage("  Nickname: " .. nick, 0.4, 0.4, 1.0);
        if (nickChannels ~= "") then
            DEFAULT_CHAT_FRAME:AddMessage("    " .. nickChannels, 0.4, 1.0, 0.4);
        else
            DEFAULT_CHAT_FRAME:AddMessage("    All channels disabled", 1.0, 0.4, 0.4);
        end
    else
        DEFAULT_CHAT_FRAME:AddMessage("  No nickname", 1.0, 0.4, 0.4);
    end
    DEFAULT_CHAT_FRAME:AddMessage("----", 1.0, 1.0, 1.0);
end

-- Displays the help information
function Identity_PrintHelp(cmdName)
    DEFAULT_CHAT_FRAME:AddMessage("----", 1.0, 1.0, 1.0);
    DEFAULT_CHAT_FRAME:AddMessage("Identity " .. Identity_VERSION .. " help:", 0.4, 0.4, 1.0);
    DEFAULT_CHAT_FRAME:AddMessage("See Readme.txt for more detailed information", 0.4, 0.4, 1.0);
    DEFAULT_CHAT_FRAME:AddMessage("/id - Displays the current configuration", 0.4, 0.4, 1.0);
    DEFAULT_CHAT_FRAME:AddMessage("/id config - Displays the current configuration", 0.4, 0.4, 1.0);
    DEFAULT_CHAT_FRAME:AddMessage("/id help - Displays this help text", 0.4, 0.4, 1.0);
    DEFAULT_CHAT_FRAME:AddMessage("/id off - Disables Identity", 0.4, 0.4, 1.0);
    DEFAULT_CHAT_FRAME:AddMessage("/id on - Enables Identity", 0.4, 0.4, 1.0);
    DEFAULT_CHAT_FRAME:AddMessage("/id main <name> - Sets the name of your main", 0.4, 0.4, 1.0);
    DEFAULT_CHAT_FRAME:AddMessage("/id nick <name> - Sets your group short name", 0.4, 0.4, 1.0);
    DEFAULT_CHAT_FRAME:AddMessage("/id enable <channels> - These channels will print your Identity", 0.4, 0.4, 1.0);
    DEFAULT_CHAT_FRAME:AddMessage("/id disable <channels> - These channels will not print your Identity", 0.4, 0.4, 1.0);
    DEFAULT_CHAT_FRAME:AddMessage("/id zone on|off - Toggles your location in your Identity", 0.4, 0.4, 1.0);
    DEFAULT_CHAT_FRAME:AddMessage("/id reset - Clears your Identity settings", 0.4, 0.4, 1.0);
    DEFAULT_CHAT_FRAME:AddMessage("----", 1.0, 1.0, 1.0);
end

-- Enables/disables Identity
function Identity_Enable(enable)
    if (enable) then
        DEFAULT_CHAT_FRAME:AddMessage("Identity enabled", 0.4, 1.0, 0.4);
    else
        DEFAULT_CHAT_FRAME:AddMessage("Identity disabled", 1.0, 0.4, 0.4);
    end
    IdentitySettings.Enabled = enable;
end

-- Checks the specified name
function Identity_CheckName(old, new)
    -- Check that a new name specified
    if (not new) then
        -- Keep the existing name
        return old;
    elseif (new == "") then
        -- Clear the name
        DEFAULT_CHAT_FRAME:AddMessage("Cleared name", 0.4, 0.4, 1.0);
        return new;
    else
        -- Update the name
        DEFAULT_CHAT_FRAME:AddMessage("Name changed from '" .. old .. "' to '" .. new .. "'", 0.4, 0.4, 1.0);
        return new;
    end
end

-- Enables/disables a list of channels
function Identity_EnableChannels(channels, enable)
    -- Check that a channel list was specified
    if (not channels or channels == "") then
        DEFAULT_CHAT_FRAME:AddMessage("No Identity channels specified", 1.0, 0.4, 0.4);
        return;
    end;

    -- Iterate over the list of channels
    for channel in string.gmatch(channels, "%w+") do
        Identity_EnableChannel(channel, enable);
    end
end

-- Enables/disables the specified channel
function Identity_EnableChannel(channel, enable)
    -- Update the channel setting and get the channel's canonical name
    local channelName = "";
    if (channel == "g" or channel == "guild") then
        channelName = "Guild";
        IdentitySettings.Channels.Guild = enable;
    elseif (channel == "o" or channel == "officer") then
        channelName = "Officer";
        IdentitySettings.Channels.Officer = enable;
    elseif (channel == "r" or channel == "raid") then
        channelName = "Raid";
        IdentitySettings.Channels.Raid = enable;
    elseif (channel == "p" or channel == "party") then
        channelName = "Party";
        IdentitySettings.Channels.Party = enable;
    elseif (channel == "w" or channel == "whisper" or channel == "t" or channel == "tell") then
        channelName = "Whispers";
        IdentitySettings.Channels.Tell = enable;
    elseif (string.match(channel, "%d+")) then
        channelName = channel;
        local chanID = string.format("C%02d", channel + 0);
        IdentitySettings.Channels[chanID] = enable;
    else
        DEFAULT_CHAT_FRAME:AddMessage("Invalid Identity channel '" .. channel .. "'", 1.0, 0.4, 0.4);
    end

    -- Print the channel's new state
    if (enable) then
        DEFAULT_CHAT_FRAME:AddMessage("Channel " .. channelName .. " enabled", 0.4, 1.0, 0.4);
    else
        DEFAULT_CHAT_FRAME:AddMessage("Channel " .. channelName .. " disabled", 1.0, 0.4, 0.4);
    end
end

-- Enables/disables zone display
function Identity_SetZoneDisplay(options)
    if (options == "on") then
        DEFAULT_CHAT_FRAME:AddMessage("Zone display enabled", 0.4, 0.4, 1.0);
        IdentitySettings.DisplayZone = true;
    elseif (options == "off") then
        DEFAULT_CHAT_FRAME:AddMessage("Zone display disabled", 0.4, 0.4, 1.0);
        IdentitySettings.DisplayZone = false;
    elseif (options == "") then
        if (IdentitySettings.DisplayZone) then
            DEFAULT_CHAT_FRAME:AddMessage("Zone display disabled", 0.4, 0.4, 1.0);
        else
            DEFAULT_CHAT_FRAME:AddMessage("Zone display enabled", 0.4, 0.4, 1.0);
        end
        IdentitySettings.DisplayZone = not IdentitySettings.DisplayZone;
    else
        DEFAULT_CHAT_FRAME:AddMessage("Invalid Identity zone argument '" .. options .. "'", 1.0, 0.4, 0.4);
    end
end

-- Sets the format string to be used for name display
function Identity_SetFormat(options)
    -- Check for no argument
    if (not options or options == "") then
        IdentitySettings.Format = "[%s]";
        DEFAULT_CHAT_FRAME:AddMessage("Reset format to '" .. IdentitySettings.Format .. "'", 0.4, 0.4, 1.0);
        return;
    end

    -- Check for the %s replacement token
    if (string.match(options, "%%s")) then
        -- Escape any other percent sign, to avoid contaminating the format string
        options = string.gsub(options, "%%([^s])", "%%%%%1");
        DEFAULT_CHAT_FRAME:AddMessage("Changed format to '" .. options .. "'", 0.4, 0.4, 1.0);
        IdentitySettings.Format = options;
    else
        DEFAULT_CHAT_FRAME:AddMessage("Invalid Identity format string '" .. options .. "'", 1.0, 0.4, 0.4);
    end
end

-- Formats the main name for prepending to a chat message
function Identity_GenerateMainName()
    local name = IdentitySettings.MainName;

    -- Check if we need to append zone information as well
    if (IdentitySettings.DisplayZone) then
        name = name .. ", " .. GetZoneText();
    end

    return string.format(IdentitySettings.Format, name);
end

-- Formats the nickname for prepending to a chat message
function Identity_GenerateNickName()
    local name = IdentitySettings.NickName;

    -- Check if we need to append zone information as well
    if (IdentitySettings.DisplayZone) then
        name = name .. ", " .. GetZoneText();
    end

    return string.format(IdentitySettings.Format, name);
end

-- Concatenates of list of enabled channels for the main character's name
function Identity_GetMainChannels()
    local channels = "";
    if (IdentitySettings.Channels.Guild) then
        channels = channels .. "Guild ";
    end
    if (IdentitySettings.Channels.Officer) then
        channels = channels .. "Officer ";
    end
    if (IdentitySettings.Channels.Tell) then
        channels = channels .. "Whisper ";
    end
    for i = 1, 10 do
        local chanID = string.format("C%02d", i);
        if (IdentitySettings.Channels[chanID]) then
            channels = channels .. i .. " ";
        end
    end
    return channels;
end

-- Concatenates of list of enabled channels for the character's nickname
function Identity_GetNickChannels()
    local channels = "";
    if (IdentitySettings.Channels.Raid) then
        channels = channels .. "Raid ";
    end
    if (IdentitySettings.Channels.Party) then
        channels = channels .. "Party ";
    end
    return channels;
end


locales/ptBR.lua
local L = LibStub("AceLocale-3.0"):NewLocale("Identity2", "ptBR", false)

if L then
    L["migration.created_new_profile"] = function(profile_name) return "Novo Perfil |cFFC6A15B" .. profile_name .. "|r criado pela migração de versão antiga" end
    L["migration.finished"] = function(created_new_profile) local text = "Migração das configurações desta personagem concluída"; if(created_new_profile) then text = text .. ". Um novo perfil foi criado por razão de conflito com o perfil ativo" end return text end
    
    L["channel.GUILD.name"] = "Bate-papo da guilda"
    L["channel.OFFICER.name"] = "Bate-papo de oficiais"
    L["channel.RAID.name"] = "Raide"
    L["channel.PARTY.name"] = "Grupo"
    L["channel.INSTANCE_CHAT.name"] = "Instância"
    L["channel.WHISPER.name"] = "Sussurrar"
    L["channel.SAY.name"] = "Dizer"
    L["channel.YELL.name"] = "Gritar"
    L["channel.BN_WHISPER.name"] = "Sussurro da Battle.net"
    
    L["options.name"] = "Opções Gerais"
    L["options.version.name"] = function(version) return "Identity 2 versão " .. version end
    L["options.enable.name"] = "Ativar"
    L["options.enable.desc"] = "Ativa/Desativa o Identity"
    L["options.fun.name"] = "Diversão"
    L["options.fun.desc"] = "Informa para o Identity se você gosta de diversão em dias especiais"
    L["options.format.name"] = "Formato"
    L["options.format.desc"] = "Define o formato usado para mostrar sua Identidade. Padrão: [%s]\n\nTokens válidos para uso no formato:\n    %s -> Será substituído pela identidade\n\n    %z -> Será substituído pelo nome da zona atual\n\n    %l -> Será substituído pelo nível da personagem\n\n    %g -> Será substituído pelo nome da Guilda da personagem\n\n    %r -> Será substituído pelo nome do Reino da personagem"
    L["options.identity.name"] = "Identidade"
    L["options.identity.desc"] = "Define o texto usado como sua Identidade"
    L["options.default_channels.name"] = "Canais Padrões"
    L["options.custom_channels.name"] = "Canais Customizados"
    L["options.custom_channels.add.name"] = "Adiciona Canal"
    L["options.custom_channels.add.desc"] = "Digite o nome do canal a ser adicionado e pressione Ok"
    L["options.custom_channels.error.empty"] = "Nome do Canal não pode ser em branco"
    L["options.custom_channels.error.already_exists"] = "Um Canal com esse nome já está cadastrado"
    L["options.communities.name"] = "Comunidades"
    L["options.communities.add.name"] = "Adiciona Comunidade"
    L["options.communities.add.desc"] = "Digite o nome da comunidade a ser adicionada e pressione Ok"
    L["options.communities.no_streams"] = "Nenhum canal nesta comunidade no momento ou o personagem não está nesta comunidade"
    L["options.communities.error.empty"] = "Nome da Comunidade não pode ser em branco"
    L["options.communities.error.already_exists"] = "Uma Comunidade com esse nome já está cadastrada"
    L["options.communities.remove.name"] = "Remover Comunidade"
    L["options.community.enable.name"] = "Ativar Comunidade"
    L["options.community.enable.desc"] = function(community) return "Ativa/Desativa " .. community end
    L["options.channels.enable.name"] = "Ativar Canal"
    L["options.channels.enable.desc"] = function(channel) return "Ativa/Desativa " .. channel end
    L["options.channels.identity.name"] = "Identidade no Canal"
    L["options.channels.identity.desc"] = function(channel, main_identity) return "Define o texto para uso em " .. channel .. " no lugar de " .. main_identity end
    L["options.channels.preview_header"] = "Prévia da Identidade"
    L["options.channels.preview.message"] = "Isto é uma prévia da Identidade atual"
    L["options.channels.remove.name"] = "Remover Canal"
    
    L["profiles.name"] = "Perfis"
    
    L["initialization.loaded"] = function(version) return "Versão " .. version .. " carregada." end
    L["initialization.updated"] = function(version) return "Atualizado para versão " .. version end

    L["fun.prank.jenkins"] = " Jenkins"
    L["fun.prank.guldan_start"] = "Gul'"
    L["fun.prank.guldan_end"] = "'Dan"
    L["fun.prank.the_cute"] =  function() return GetRandomArgument(" o Fofo", " a Fofa") end
    L["fun.prank.magnificient"] = "Magnificente "
    L["fun.prank.the_third"] = function(identity) return GetRandomArgument("Senhor ", "Senhora ") .. identity .. " III" end
    L["fun.prank.prince"] = "Aquele anteriormente conhecido como "
    L["fun.prank.master_roshi"] = "Mestre Kame"
    L["fun.prank.whats_the_name_of_the_song"] = "Darude - Sandstorm"
    L["fun.prank.404"] = "#INSIRA_INDENTIDADE_AQUI#"
    L["fun.prank.univere_life_everything_else"] = "42"
    L["fun.prank.rhonin_best_quote"] = function() return GetRandomArgument("Cidadão", "Cidadã") .. " de Dalaran" end
    L["fun.prank.not"] = "Não "
    L["fun.prank.plated"] = function(identity) return identity .. " Banhado à " .. GetRandomArgument("ouro", "prata", "cobre") end
    L["fun.prank.item_quality"] = function(identity) return GetRandomArgument("Artifact(Gold) ", "Heirloom(Cyan) ", "Legendary(Orange) ", "Epic(Purple) ", "Rare(Blue) ", "Uncommon(Green) ", "Common(White) ", "Poor(Grey) ")  .. identity end
    L["fun.prank.size"] = function() return GetRandomArgument("Pequeno ", "Grande ") end
    
    L["fun.hohoho.santa"] = function(identity) return identity .. " Noel" end --usar Papai ou Mamãe sem o Noel fica muito estranho
    L["fun.hohoho.claus"] = function(identity) return identity .. " Noel" end
    L["fun.hohoho.red_nose"] = function(identity) return identity .. ", do Nariz Vermelho" end
end


locales/enUS.lua
local L = LibStub("AceLocale-3.0"):NewLocale("Identity2", "enUS", true)

if L then
    L["migration.created_new_profile"] = function(profile_name) return "New profile |cFFC6A15B" .. profile_name .. "|r created by migration of old version" end
    L["migration.finished"] = function(created_new_profile) local text = "Migration of this character's configurations finished"; if(created_new_profile) then text = text .. ". A new profile was created because of conflicts with the active profile" end return text end
    
    L["channel.GUILD.name"] = "Guild"
    L["channel.OFFICER.name"] = "Officer"
    L["channel.RAID.name"] = "Raid"
    L["channel.PARTY.name"] = "Party"
    L["channel.INSTANCE_CHAT.name"] = "Instance"
    L["channel.WHISPER.name"] = "Whisper"
    L["channel.SAY.name"] = "Say"
    L["channel.YELL.name"] = "Yell"
    L["channel.BN_WHISPER.name"] = "Battle.net Whisper"
    
    L["options.name"] = "General Options"
    L["options.version.name"] = function(version) return "Identity 2 version " .. version end
    L["options.enable.name"] = "Enable"
    L["options.enable.desc"] = "Enable/Disable Identity"
    L["options.fun.name"] = "Fun"
    L["options.fun.desc"] = "Inform Identity if you like to have fun on special days"
    L["options.format.name"] = "Format"
    L["options.format.desc"] = "Sets the format used to display your Identity. Default: [%s]\n\nValid tokens for use in the format:\n    %s -> Will be replaced by the identity\n\n    %z -> Will be replaced by the name of the current zone\n\n    %l -> Will be replaced by the character level\n\n    %g -> Will be replaced by the character guild\n\n    %r -> Will be replaced by the realm name\n\n    %f -> Will be replaced by the character faction"
    L["options.identity.name"] = "Identity"
    L["options.identity.desc"] = "Sets the string used as your Identity"
    L["options.default_channels.name"] = "Default Channels"
    L["options.custom_channels.name"] = "Custom Channels"
    L["options.custom_channels.add.name"] = "Add Channel"
    L["options.custom_channels.add.desc"] = "Type the name of the channel to add and press Okay"
    L["options.custom_channels.error.empty"] = "Channel name must not be empty"
    L["options.custom_channels.error.already_exists"] = "A Channel with this name is already registered"
    L["options.communities.name"] = "Communities"
    L["options.communities.error.empty"] = "Community name must not be empty"
    L["options.communities.error.already_exists"] = "A Community with this name is already registered"
    L["options.communities.add.name"] = "Add Community"
    L["options.communities.add.desc"] = "Type the name of the community to add and press Okay"
    L["options.communities.no_streams"] = "No channels on this community at the moment or the character is not in this community"
    L["options.communities.remove.name"] = "Remove Community"
    L["options.community.enable.name"] = "Enable Community"
    L["options.community.enable.desc"] = function(community) return "Enable/Disable " .. community end
    L["options.channels.enable.desc"] = function(channel) return "Enable/Disable " .. channel end
    L["options.channels.enable.name"] = "Enable Channel"
    L["options.channels.identity.name"] = "Channel Identity"
    L["options.channels.identity.desc"] = function(channel, main_identity) return "Sets a string to use in " .. channel .. " instead of " .. main_identity end
    L["options.channels.preview_header"] = "Identity Preview"
    L["options.channels.preview.message"] = "This is a preview of the current Identity"
    L["options.channels.remove.name"] = "Remove Channel"
    
    L["profiles.name"] = "Profiles"
    
    L["initialization.loaded"] = function(version) return "Version " .. version .. " loaded." end
    L["initialization.updated"] = function(version) return "Updated to version " .. version end
    
    L["fun.prank.jenkins"] = " Jenkins"
    L["fun.prank.guldan_start"] = "Gul'"
    L["fun.prank.guldan_end"] = "'Dan"
    L["fun.prank.the_cute"] =  function() return " the Cute" end
    L["fun.prank.magnificient"] = "Magnificient "
    L["fun.prank.the_third"] = function(identity) return GetRandomArgument("Lord ", "Lady ") .. identity .. " the III" end
    L["fun.prank.prince"] = "The one formerly known as "
    L["fun.prank.master_roshi"] = "Master Roshi"
    L["fun.prank.whats_the_name_of_the_song"] = "Darude - Sandstorm"
    L["fun.prank.404"] = "#INSERT_IDENTITY_HERE#"
    L["fun.prank.univere_life_everything_else"] = "42"
    L["fun.prank.rhonin_best_quote"] = function() return "Citizen of Dalaran" end
    L["fun.prank.not"] = "Not "
    L["fun.prank.plated"] = function(identity) return GetRandomArgument("Gold", "Silver", "Cooper") .. " plated " .. identity end
    L["fun.prank.item_quality"] = function(identity) return GetRandomArgument("Artefato(Dourado) ", "Herança(Ciano) ", "Lendário(Laranja) ", "Épico(Roxo) ", "Raro(Azul) ", "Incomum(Verde) ", "Comum(Branco) ", "Inferior(Cinza) ")  .. identity end
    L["fun.prank.size"] = function() return GetRandomArgument("Small ", "Big ") end
    
    L["fun.hohoho.santa"] = function(identity) return "Santa " .. identity end
    L["fun.hohoho.claus"] = function(identity) return identity .. " Claus" end
    L["fun.hohoho.red_nose"] = function(identity) return identity .. ", the Red-Nosed" end
end
